usr/src/psm/stand/boot/aarch64/common/arch_timer.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+ * or http://www.opensolaris.org/os/licensing.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+#include <sys/types.h>
+#include <sys/promif.h>
+#include <sys/controlregs.h>
+#include <sys/arch_timer.h>
+
+/* Generic Timer control bits */
+#define CNT_CTL_ENABLE  (1u << 0)
+#define CNT_CTL_IMASK   (1u << 1)
+#define CNT_CTL_ISTATUS (1u << 2)
+
+struct arch_timer_ops {
+	void (*write_cnt_cval)(uint64_t);
+	uint64_t (*read_cnt_ctl)(void);
+	void (*write_cnt_ctl)(uint64_t);
+	uint64_t (*read_cnt_ct)(void);
+};
+
+static struct arch_timer_ops timer_ops[] = {
+	{ /* phys timer */
```

Comment:
can we use C99 array init so we have `[TMR_PHYS] = ...` and `[TMR_VIRT] = ...` and not need the comments?

Reply:
Done.